### PR TITLE
Smithy 1.23 changelog

### PR DESCRIPTION
**Pull request by DavidOgunsAWS**
*Description of changes:*
* Preparing for upcoming release. VERSION file update along with changelog entries.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,97 @@
...
+
+### Features
+
+* Added version 2.0 of the Smithy IDL with many enhancements to the modeling experience. ([#1317](https://github.com/awslabs/smithy/pull/1317),
+  [#1312](https://github.com/awslabs/smithy/pull/1312), [#1318](https://github.com/awslabs/smithy/pull/1318))
+* Added mixins for all shape types. ([#889](https://github.com/awslabs/smithy/pull/889), [#1025](https://github.com/awslabs/smithy/pull/1025),
+  [#1139](https://github.com/awslabs/smithy/pull/1139), [#1323](https://github.com/awslabs/smithy/pull/1323))
+* Added resource properties to version 2.0 of Smithy IDL. ([#1213](https://github.com/awslabs/smithy/pull/1213))
+* Added target elision for mixins/resources. ([#1231](https://github.com/awslabs/smithy/pull/1231))
+* Added inline operation IO shapes. ([#963](https://github.com/awslabs/smithy/pull/963), [#962](), [#1007](https://github.com/awslabs/smithy/pull/1007))
```
Comment (syall) [inline CHANGELOG.md:13]: `#962` is missing the PR link: `https://github.com/awslabs/smithy/pull/962`

**Review by syall — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,97 @@
...
+* Added validation for multiple IDL versions. ([#917](https://github.com/awslabs/smithy/pull/917))
+* Added IDL 1.0 to 2.0 model migration tool. ([#1175](https://github.com/awslabs/smithy/pull/1175))
+* Added enum shapes. ([#1088](https://github.com/awslabs/smithy/pull/1088), [#1114](https://github.com/awslabs/smithy/pull/1114),
+  [#1133](https://github.com/awslabs/smithy/pull/1133), [#1313](https://github.com/awslabs/smithy/pull/1313))
+* Added `@clientOptional` trait. ([#1052](https://github.com/awslabs/smithy/pull/1052), [#1264](https://github.com/awslabs/smithy/pull/1264))
+* Multiple traits can now be allpied in a single `apply` statement. ([#885](https://github.com/awslabs/smithy/pull/885))
```
Comment (syall) [inline CHANGELOG.md:19]: Typo: `allpied` should be `applied`

**Review by syall — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (DavidOgunsAWS) [reply CHANGELOG.md:19]: Far less fun though

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,98 @@
...
+  [#1007](https://github.com/awslabs/smithy/pull/1007))
+* Added validation for multiple IDL versions. ([#917](https://github.com/awslabs/smithy/pull/917))
+* Added IDL 1.0 to 2.0 model migration tool. ([#1175](https://github.com/awslabs/smithy/pull/1175))
+* Added enum shapes. ([#1088](https://github.com/awslabs/smithy/pull/1088), [#1114](https://github.com/awslabs/smithy/pull/1114),
+  [#1133](https://github.com/awslabs/smithy/pull/1133), [#1313](https://github.com/awslabs/smithy/pull/1313))
+* Added `@clientOptional` trait. ([#1052](https://github.com/awslabs/smithy/pull/1052), [#1264](https://github.com/awslabs/smithy/pull/1264))
+* Multiple traits can now be applied in a single `apply` statement. ([#885](https://github.com/awslabs/smithy/pull/885))
+* Commas are now optional in the IDL. ([#772](https://github.com/awslabs/smithy/pull/772), [#776](https://github.com/awslabs/smithy/pull/776),
+  [#1166](https://github.com/awslabs/smithy/pull/1166))
+* `@default` and `@required` traits added. ([#1019](https://github.com/awslabs/smithy/pull/1019), [#1286](https://github.com/awslabs/smithy/pull/1286),
```
Comment (syall) [inline CHANGELOG.md:23]: Change this to: Added `@default` and `@required` traits.

**Review by syall — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,98 @@
...
+  [#1007](https://github.com/awslabs/smithy/pull/1007))
+* Added validation for multiple IDL versions. ([#917](https://github.com/awslabs/smithy/pull/917))
+* Added IDL 1.0 to 2.0 model migration tool. ([#1175](https://github.com/awslabs/smithy/pull/1175))
+* Added enum shapes. ([#1088](https://github.com/awslabs/smithy/pull/1088), [#1114](https://github.com/awslabs/smithy/pull/1114),
+  [#1133](https://github.com/awslabs/smithy/pull/1133), [#1313](https://github.com/awslabs/smithy/pull/1313))
+* Added `@clientOptional` trait. ([#1052](https://github.com/awslabs/smithy/pull/1052), [#1264](https://github.com/awslabs/smithy/pull/1264))
+* Multiple traits can now be applied in a single `apply` statement. ([#885](https://github.com/awslabs/smithy/pull/885))
+* Commas are now optional in the IDL. ([#772](https://github.com/awslabs/smithy/pull/772), [#776](https://github.com/awslabs/smithy/pull/776),
+  [#1166](https://github.com/awslabs/smithy/pull/1166))
+* Added `@default` and `@required` traits. ([#1019](https://github.com/awslabs/smithy/pull/1019), [#1286](https://github.com/awslabs/smithy/pull/1286),
```
Comment (mtdowling) [inline CHANGELOG.md:23]: We didn't add the `@required` trait

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,98 @@
 # Smithy Changelog
 
+## 1.23.0 (2022-08-10)
+
+### Features
+
+* Added version 2.0 of the Smithy IDL with many enhancements to the modeling experience. ([#1317](https://github.com/awslabs/smithy/pull/1317),
```
Comment (mtdowling) [inline CHANGELOG.md:7]:
```suggestion
* Added version 2.0 of the Smithy IDL. ([#1317](https://github.com/awslabs/smithy/pull/1317),
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,98 @@
...
+
+### Documentation
+
+* Added documentation for IDL 2.0 and changed location of 1.0 docs. ([#1302](https://github.com/awslabs/smithy/pull/1302),
+  [#1057](https://github.com/awslabs/smithy/pull/1057))
+* Added IDL 1.0 to 2.0 migration guide. ([#1065](https://github.com/awslabs/smithy/pull/1065), [#1074](https://github.com/awslabs/smithy/pull/1074))
+* Updated the doc highlighter for IDL 2.0. ([#1251](https://github.com/awslabs/smithy/pull/1251))
+* Added documentation to IAM trait enums. ([#1322](https://github.com/awslabs/smithy/pull/1322))
+
+## 1.22.0 (2022-06-30)
```
Comment (mtdowling) [inline CHANGELOG.md:47]: This is strange

**Review by mtdowling — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 6 above)
Comment (DavidOgunsAWS) [reply CHANGELOG.md]: I removed it after picking it up when the PR was initially made against main branch.

Comment (syall) [reply CHANGELOG.md]: Just to clarify:

- `1.22.0` is on the `main` branch
- `1.22.0` is not on the `idl-2.0` branch
- `1.21.0` is on the `idl-2.0` branch
- `1.23.0` is on the `idl-2.0` branch

When we merge, we will have to resolve conflicts due to the missing `1.22.0` on the `idl-2.0` branch.

**Comment by syall** [comment]
Looking at the latest commit for `idl2-changelog`, it looks like the 1.22.0 release notes are removed.

We should add them back, especially since `idl-2.0` has the 1.22.0 release notes.

**Pull request merged**